### ISSUE TITLE
Playwright: Sidebar continues to be flakey

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#56212]
### Details

There are multiple instances of sidebar continuing to be flakey despite all the work put into it.

Examples:
- https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_desktop/6687959
- https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_desktop/6692053
- https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/6692195
- https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/6692204

A large part of this is likely due to the recent changes to viewport which has made the tests less reliable due to shift in elements that uproot assumptions that were held when tests were being written.

**Comment by worldomonation** [comment]
I think we should look at a more programmatical solution that we dismissed early on: `dispatchEvent`.

Originally, the argument against using `dispatchEvent` was that 'it is not truly representative of user action'. However, seeing the amount of time and effort spent on trying to make a generic Sidebar method work (and not quite succeeding), I think it's time to re-evaluate this solution.

I've also evaluated `page.click( force=true)`, but I `dispatchEvent` has some advantages:

- works on both mobile and desktop viewports.
- eliminates dependency on where the element is in the viewport.

The second point (about visibility/location in the vievwport) is quite important.
Using the standard desktop sized viewport, logging into Atomic sites would produce a sidebar that extends past the bottom edge of the viewport and thus would have to be scrolled. This is typically an issue for elements such as `Settings` but can also affect elements at the top of the Sidebar (if the sidebar is scrolled all the way down).

Using `PageUp` and `PageDown` to bring the elements into view adds a lot of unnecessary complexity to this code that can break again when the viewport changes, or the Sidebar behavior alters. `ElementHandle.scrollIntoViewIfNeeded` has already been demonstrated to not work due to the 'sticky' nature of the Sidebar. Implementing our own `scrollIfNeeded` using JavaScript (current solution) does not work reliably if the Sidebar shifts during load.

I think we shouldn't discount using all of the tools that we're provided with, and `dispatchEvent` is there exactly for these sorts of reasons.
